client/src/containers/SubmitCodeSmells.js

```diff
@@ -0,0 +1,118 @@
+import React, { PropTypes, Component } from 'react';
+import { connect } from 'react-redux';
+import { selectCode, submitCode } from '../actions/code.js';
+import CodeBlock from '../components/CodeBlock';
+
+class SubmitCodeSmells extends Component {
+	constructor(props) {
+		super(props);
+		this.clickAction = this.clickAction.bind(this);
+		this.handleSubmit = this.handleSubmit.bind(this);
+		this.selectCodeSmell = this.selectCodeSmell.bind(this);
+		this.state = {
+			codeSmellName: ""
+		};
+	}
+
+	clickAction(lineNumber) {
+        const { dispatch } = this.props;
+        if (this.state.codeSmellName !== "") {
+            dispatch(selectCode(lineNumber, this.state.codeSmellName));
+        }
+    }
+
+	handleSubmit() {
+		const { dispatch, code, userid, selectedLines } = this.props;
+		dispatch(submitCode(userid, code, selectedLines));
+		
+		var codeid = localStorage.getItem("codeid"),
```

Comment:
You might think about adding action creators and saving the codeid into the application state. It might take more time to code but it's a little more proper.